google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/stub/mutaterows/MutateRowsBatchingDescriptorTest.java

```diff
@@ -92,11 +96,54 @@ public void splitResponseTest() {
     assertThat(batchResponse.get(1).getResultFuture().isDone()).isFalse();
 
     MutateRowsBatchingDescriptor underTest = new MutateRowsBatchingDescriptor();
-    underTest.splitResponse(null, batchResponse);
+    underTest.splitResponse(new MutateRowsAttemptResult(), batchResponse);
     assertThat(batchResponse.get(0).getResultFuture().isDone()).isTrue();
     assertThat(batchResponse.get(1).getResultFuture().isDone()).isTrue();
   }
 
+  @Test
+  public void splitResponsePartialErrorsTest() throws Exception {
+    BatchEntry<RowMutationEntry, Void> batchEntry1 =
+        BatchEntry.create(
+            RowMutationEntry.create("key1").deleteRow(), SettableApiFuture.<Void>create());
+    BatchEntry<RowMutationEntry, Void> batchEntry2 =
+        BatchEntry.create(
+            RowMutationEntry.create("key2").deleteRow(), SettableApiFuture.<Void>create());
+
+    List<BatchEntry<RowMutationEntry, Void>> batchResponse =
+        ImmutableList.of(batchEntry1, batchEntry2);
+    assertThat(batchResponse.get(0).getResultFuture().isDone()).isFalse();
+    assertThat(batchResponse.get(1).getResultFuture().isDone()).isFalse();
+
+    MutateRowsBatchingDescriptor underTest = new MutateRowsBatchingDescriptor();
+    underTest.splitResponse(
+        new MutateRowsAttemptResult(
+            Arrays.asList(
+                FailedMutation.create(
+                    0,
+                    ApiExceptionFactory.createException(
+                        "error message",
+                        null,
+                        GrpcStatusCode.of(io.grpc.Status.Code.INTERNAL),
+                        false))),
+            true),
+        batchResponse);
+    assertThat(batchResponse.get(0).getResultFuture().isDone()).isTrue();
+    assertThat(batchResponse.get(1).getResultFuture().isDone()).isTrue();
+
+    batchResponse.get(1).getResultFuture().get();
+
+    Throwable actualError = null;
+    try {
+      batchResponse.get(0).getResultFuture().get();
+    } catch (Throwable t) {
+      actualError = t.getCause();
+    }
```

Comment:
This would be better:
```
Throwable actualError = assertThrows(
  ExecutionException.class,
  () -> batchResponse.get(0).getResultFuture().get()
 ).getCause();
```